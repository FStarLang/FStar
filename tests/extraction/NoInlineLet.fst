module NoInlineLet
open FStar.Tactics

let test_no_inline (f: int): int =
  (* this one shouldn't be inlined in tactics or extraction *)
  [@@no_inline_let]
  let x = f + 1 in
  (* this one will be inlined by the norm tactic in the below test *)
  let y = x + 2 in
  y + y

[@@(postprocess_with (fun _ -> norm [delta_only [`%test_no_inline]]; trefl ()))]
let test_postprocess (f: int): int =
  let x = test_no_inline f in
  x
