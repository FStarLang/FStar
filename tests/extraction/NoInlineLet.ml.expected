open Prims
let (test_no_inline : Prims.int -> Prims.int) =
  fun f ->
    let x = f + Prims.int_one in let y = x + (Prims.of_int (2)) in y + y
let (test_postprocess : Prims.int -> Prims.int) =
  fun f ->
    let x = f + Prims.int_one in
    (x + (Prims.of_int (2))) + (x + (Prims.of_int (2)))
